test/Benchmarks/src/Sieve/heap_dump_lazy_sieve.js

```diff
@@ -0,0 +1,50 @@
+//
+// Demo script showing a "serialization" of an Enso atom structure
+// to a `.hprof` file which can be opened and analyzed in VisualVM
+//
+// Based on following tutorial:
+// https://www.graalvm.org/jdk21/tools/graalvm-insight/manual/#heap-dumping
+//
+// Execute from `sbt` as:
+// ```
+// sbt:enso> runEngineDistribution \
+//    --vm.D=polyglot.heap.dump=/tmp/sieve.hprof \
+//    --vm.D=polyglot.insight=Benchmarks/src/Sieve/heap_dump_lazy_sieve.js \
+//    --run test/Benchmarks/src/Sieve/Lazy_Sieve.enso
+//    10000
+// ```
+//
+// This GraalVM script waits for the end execution of `compute_and_print_nth` function
+// to store value of its `p` local variable (containing linked list of prime numbers)
+// into `/tmp/sieve.hprof` - a GraalVM serialization for Truffle languages!
```

Comment:
This is how the heap dump looks like in VisualVM:

![Enso heap in VisualVM](https://github.com/user-attachments/assets/d0ac9c11-3830-4b9b-a8c6-20d0771c8396)